### PR TITLE
feat(vrl): Display VRL diagnostic messages in web UI

### PR DESCRIPTION
**Pull request by jonathanpv**
Closes https://github.com/vectordotdev/vector/issues/14659

Current iteration for printing out diagnostic messages while attempting to setup a more complex structure to eventually add severity, contains_warnings, etc as fields for future use. 

This will print out the diagnostic message in the existing output box but without using xterm.js as I found using the uncolorized field will satisfy the requirement and reduce complexity of introducing another library. 

If feedback is to add colorization then we can support that by adding the xterm.js console window. 

image

### CODE REVIEW

**Comment by fuchsnj** [comment]
> If feedback is to add colorization then we can support that by adding the xterm.js console window.

That seems like something we can add in the future, but is probably not as important as some of the other features we want to add.

File: lib/vrl/web-playground/src/lib.rs [hunk 1]
```diff
@@ -59,16 +82,14 @@ fn compile(mut input: Input) -> Result<VrlCompileResult, String> {
...
-            return Err(msg);
+            diagnostics_res = VrlDiagnosticResult::new(&input.program, diagnostics);
+            return Err(diagnostics_res);
         }
     };
 
     match runtime.resolve(&mut target_value, &program.program, &timezone) {
         Ok(result) => Ok(VrlCompileResult::new(result, target_value.value)),
-        Err(err) => Err(err.to_string()),
+        Err(_err) => Err(diagnostics_res),
```
Comment (fuchsnj) [inline lib/vrl/web-playground/src/lib.rs:92]: Maybe move the TODO from above `VrlDiagnosticResult` to here? It wasn't obvious at first that this error is basically being ignored. Is there an issue for this? If not we should probably make one, and include the issue id in a comment here

**Review by fuchsnj — APPROVED** [review]
One small comment, otherwise this looks good.

File: lib/vrl/web-playground/src/lib.rs (hunk 1 above)
Comment (jonathanpv) [reply lib/vrl/web-playground/src/lib.rs:89]: Tracking in https://github.com/vectordotdev/vector/issues/15128

**Pull request merged**